From 97d2850af13c339369093b78abe5265845d78220 Thu Mar 12 15:51:45 2020 +0100
From: Thomas Lea <thomas_lea@comcast.com>
Date: Mon, 21 Sep 2020 08:56:02 -0500
Subject: [PATCH] Added CMakeLists.txt
Source: COMCAST
License: BSD-2-Clause
Upstream-Status: Pending
Signed-off-by: Thomas Lea <thomas_lea@comcast.com>
---
Index: git/CMakeLists.txt
===================================================================
--- /dev/null
+++ git/CMakeLists.txt
@@ -0,0 +1,21 @@
+cmake_minimum_required(VERSION 2.8.12)
+project(linenoise)
+
+set(CMAKE_DEBUG_POSTFIX d)
+add_library(linenoise SHARED linenoise.c)
+
+target_include_directories(linenoise PUBLIC
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
+    $<INSTALL_INTERFACE:include>
+)
+
+set_property(TARGET linenoise PROPERTY VERSION "1.0")
+
+install(TARGETS linenoise
+    ARCHIVE DESTINATION lib
+    LIBRARY DESTINATION lib
+    RUNTIME DESTINATION bin
+)
+
+install(FILES linenoise.h DESTINATION include)
+
