Index: webbridge-master/CMakeLists.txt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- webbridge-master/CMakeLists.txt	(revision f2ccaa0b8e70164ca372e42aaaba71848305f2df)
+++ webbridge-master/CMakeLists.txt	(revision )
@@ -147,6 +147,7 @@
 # default switch configuration
 map()
   key(switch)
+  key(resumes)
 end()
 ans(WEBBRIDGE_PLUGIN_CONTROLLER_CONFIGURATION)
 
@@ -172,6 +173,10 @@
 if (WEBBRIDGE_PLUGIN_WEBSERVER)
   map_append(${WEBBRIDGE_PLUGIN_CONTROLLER_CONFIGURATION} resumes WebServer)
 endif (WEBBRIDGE_PLUGIN_WEBSERVER)
+
+if (WEBBRIDGE_PLUGIN_WEBKITBROWSER)
+	map_append(${WEBBRIDGE_PLUGIN_CONTROLLER_CONFIGURATION} resumes WebKitBrowser)
+endif (WEBBRIDGE_PLUGIN_WEBKITBROWSER)
 
 map_append(${WEBBRIDGE_PLUGIN_CONTROLLER_CONFIGURATION} switch ${WEBBRIDGE_PLUGIN_CONTROLLER_SWITCH})
 map_append(${WEBBRIDGE_PLUGIN_CONTROLLER} configuration ${WEBBRIDGE_PLUGIN_CONTROLLER_CONFIGURATION})
