From b2a966fe96bfde78977ddd36d5aacc0301342fe8 Mon Sep 17 00:00:00 2001
From: wouterlucas <wouter@wouterlucas.com>
Date: Thu, 15 Jun 2017 10:45:45 -0700
Subject: [PATCH] Temp workaround for unused __DEBUG__ section that throws a
 linking error

---
 WebBridge/WebBridge.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/WebBridge/WebBridge.cpp b/WebBridge/WebBridge.cpp
index 8cc5856..2186893 100644
--- a/WebBridge/WebBridge.cpp
+++ b/WebBridge/WebBridge.cpp
@@ -338,7 +338,7 @@ extern "C" {
                 }
                 #ifndef __WIN32__
                 case 'M': {
-                    #ifdef __DEBUG__
+                    #if 0
                     printf("\nMonitor callstack:\n");
                     printf("============================================================\n");
 
-- 
2.7.4

