From 76ce8e121996b797346452e74c603db4f353371f Mon Sep 17 00:00:00 2001
From: HaseenaSainul <h.sainul@metrological.com>
Date: Mon, 9 Sep 2019 11:52:21 +0530
Subject: [PATCH 1/2] ensure child count is refershed before processing
 requredChildren

---
 WebKitBrowser/WebKitImplementation.cpp | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/WebKitBrowser/WebKitImplementation.cpp b/WebKitBrowser/WebKitImplementation.cpp
index 8a9fda4..eb91c16 100644
--- a/WebKitBrowser/WebKitImplementation.cpp
+++ b/WebKitBrowser/WebKitImplementation.cpp
@@ -1299,7 +1299,7 @@ namespace WebKitBrowser {
         {
             if (pid != 0) {
                 _main = Core::ProcessInfo(pid);
-                _children = _main.Id();
+                _children = Core::ProcessInfo::Iterator(_main.Id());
                 _startTime = Core::Time::Now().Ticks() + (TYPICAL_STARTUP_TIME * 1000000);
             } else {
                 _startTime = 0;
@@ -1383,6 +1383,10 @@ namespace WebKitBrowser {
                 // In the end we check if all bits are 0, what means all mandatory processes are still running.
                 requiredProcesses = (0xFFFFFFFF >> (32 - requiredChildren));
 
+                if (_children.Count() < 2) {
+                    // Refresh the children list !!!
+                    _children = Core::ProcessInfo::Iterator(_main.Id());
+                }
                 //!< If there are less children than in the the mandatoryProcesses struct, we are done and return false.
                 if (_children.Count() >= requiredChildren) {
 
-- 
2.20.1

