From 81b29fd6403d370ec2f15b7245848c4e2627f3db Mon Sep 17 00:00:00 2001
From: wouterlucas <wouter@wouterlucas.com>
Date: Thu, 25 Jul 2019 14:49:09 -0700
Subject: [PATCH] [WebKitBrowser] Default to 1080p instead of 720p

---
 WebKitBrowser/WebKitImplementation.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/WebKitBrowser/WebKitImplementation.cpp b/WebKitBrowser/WebKitImplementation.cpp
index 8a9fda4..71d9b85 100644
--- a/WebKitBrowser/WebKitImplementation.cpp
+++ b/WebKitBrowser/WebKitImplementation.cpp
@@ -334,8 +334,8 @@ static GSourceFuncs _handlerIntervention =
                 , Automation(false)
                 , WebGLEnabled(true)
                 , ThreadedPainting()
-                , Width(1280)
-                , Height(720)
+                , Width(1920)
+                , Height(1080)
                 , PTSOffset(0)
                 , ScaleFactor(1.0)
                 , MaxFPS(60)
