From 766f41658b0dc63eafe6bef22ac3a9635153d790 Mon Sep 17 00:00:00 2001
From: HaseenaSainul <h.sainul@metrological.com>
Date: Mon, 9 Sep 2019 15:15:07 +0530
Subject: [PATCH 2/2] WebKitBrowser: magic number for required child moved to
 RequiredChildren ConstExpression

---
 WebKitBrowser/WebKitImplementation.cpp | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/WebKitBrowser/WebKitImplementation.cpp b/WebKitBrowser/WebKitImplementation.cpp
index eb91c16..c4e397d 100644
--- a/WebKitBrowser/WebKitImplementation.cpp
+++ b/WebKitBrowser/WebKitImplementation.cpp
@@ -1276,6 +1276,7 @@ namespace WebKitBrowser {
         _T("WPEWebProcess")
     };
 
+    static constexpr uint16_t RequiredChildren = (sizeof(mandatoryProcesses) / sizeof(mandatoryProcesses[0]));
     class MemoryObserverImpl : public Exchange::IMemory {
     private:
         MemoryObserverImpl();
@@ -1311,7 +1312,7 @@ namespace WebKitBrowser {
             uint32_t result(0);
 
             if (_startTime != 0) {
-                if (_children.Count() < 2) {
+                if (_children.Count() < RequiredChildren) {
                     _children = Core::ProcessInfo::Iterator(_main.Id());
                 }
 
@@ -1331,7 +1332,7 @@ namespace WebKitBrowser {
             uint32_t result(0);
 
             if (_startTime != 0) {
-                if (_children.Count() < 2) {
+                if (_children.Count() < RequiredChildren) {
                     _children = Core::ProcessInfo::Iterator(_main.Id());
                 }
 
@@ -1351,7 +1352,7 @@ namespace WebKitBrowser {
             uint32_t result(0);
 
             if (_startTime != 0) {
-                if (_children.Count() < 2) {
+                if (_children.Count() < RequiredChildren) {
                     _children = Core::ProcessInfo::Iterator(_main.Id());
                 }
 
@@ -1377,18 +1378,17 @@ namespace WebKitBrowser {
             uint32_t requiredProcesses = 0;
 
             if (_startTime != 0) {
-                const uint8_t requiredChildren = (sizeof(mandatoryProcesses) / sizeof(mandatoryProcesses[0]));
 
                 //!< We can monitor a max of 32 processes, every mandatory process represents a bit in the requiredProcesses.
                 // In the end we check if all bits are 0, what means all mandatory processes are still running.
-                requiredProcesses = (0xFFFFFFFF >> (32 - requiredChildren));
+                requiredProcesses = (0xFFFFFFFF >> (32 - RequiredChildren));
 
-                if (_children.Count() < 2) {
+                if (_children.Count() < RequiredChildren) {
                     // Refresh the children list !!!
                     _children = Core::ProcessInfo::Iterator(_main.Id());
                 }
                 //!< If there are less children than in the the mandatoryProcesses struct, we are done and return false.
-                if (_children.Count() >= requiredChildren) {
+                if (_children.Count() >= RequiredChildren) {
 
                     _children.Reset();
 
@@ -1398,13 +1398,13 @@ namespace WebKitBrowser {
                         uint8_t count(0);
                         string name(_children.Current().Name());
 
-                        while ((count < requiredChildren) && (name != mandatoryProcesses[count])) {
+                        while ((count < RequiredChildren) && (name != mandatoryProcesses[count])) {
                             ++count;
                         }
 
                         //<! this is a mandatory process and if its still active reset its bit in requiredProcesses.
                         //   If not we are not completely operational.
-                        if ((count < requiredChildren) && (_children.Current().IsActive() == true)) {
+                        if ((count < RequiredChildren) && (_children.Current().IsActive() == true)) {
                             requiredProcesses &= (~(1 << count));
                         }
                     }
-- 
2.20.1

