diff --git a/examples/CMakeLists.txt b/examples/CMakeLists.txt
index 89c5cb5..c07ba35 100644
--- a/examples/CMakeLists.txt
+++ b/examples/CMakeLists.txt
@@ -3,7 +3,7 @@ cmake_minimum_required(VERSION 3.3)
 option(PLUGIN_JSONRPC "Include JSONRPCExamplePlugin plugin" OFF)
 option(PLUGIN_TESTUTILITY "Include TestUtility plugin" OFF)
 option(PLUGIN_TESTCONTROLLER "Include TestController plugin" OFF)
-option(PLUGIN_FILETRANSFER "Include FileTransfer plugin" OFF)
+option(PLUGIN_TESTFILETRANSFER "Include FileTransfer plugin" OFF)
 option(PLUGIN_OUTOFPROCESS "Include OutOfProcess plugin" OFF)
 
 if(PLUGIN_TESTUTILITY)
@@ -19,7 +19,7 @@ if(PLUGIN_JSONRPC)
     add_subdirectory(JSONRPCClient)
 endif()
 
-if(PLUGIN_FILETRANSFER)
+if(PLUGIN_TESTFILETRANSFER)
     add_subdirectory(FileTransferClient)
 endif()
 
diff --git a/examples/FileTransferClient/FileTransferClient.cpp b/examples/FileTransferClient/FileTransferClient.cpp
index 667669d..8e6d54f 100644
--- a/examples/FileTransferClient/FileTransferClient.cpp
+++ b/examples/FileTransferClient/FileTransferClient.cpp
@@ -194,7 +194,7 @@ int main(int argc, char** argv)
 
         Core::File configFile("FileTransferClient.json", false);
         if (configFile.Exists() && configFile.Open(true)) {
-            config.FromFile(configFile);
+            config.IElement::FromFile(configFile);
             printf(_T("Get configuration from config file\n"));
         } else {
             printf(_T("Wrong parameter, use option -h to check required input arguments\n"));
