commit 60ed38b461b777c6f8623328718feb77c4751e16
Author: HaseenaSainul <h.sainul@metrological.com>
Date:   Wed Feb 12 17:27:35 2020 +0530

    Streamer: fix to avoid crash during the stop

diff --git a/Streamer/Frontend.h b/Streamer/Frontend.h
index 774861e..b2b2b54 100644
--- a/Streamer/Frontend.h
+++ b/Streamer/Frontend.h
@@ -78,6 +78,10 @@ namespace Player {
                 ~DecoderImplementation() override
                 {
                     _parent.Detach();
+                    if (_callback != nullptr) {
+                        _callback->Release();
+                        _callback = nullptr;
+                    }
                 }
 
             public:
@@ -88,9 +92,6 @@ namespace Player {
                 uint32_t Release() const override
                 {
                     if (Core::InterlockedDecrement(_referenceCount) == 0) {
-                        if (_callback != nullptr) {
-                            _callback->Release();
-                        }
                         delete this;
                         return (Core::ERROR_DESTRUCTION_SUCCEEDED);
                     }
