diff --git a/Cobalt/CMakeLists.txt b/Cobalt/CMakeLists.txt
index ad459c2..ba2d1b2 100644
--- a/Cobalt/CMakeLists.txt
+++ b/Cobalt/CMakeLists.txt
@@ -61,5 +61,18 @@ install(TARGETS ${MODULE_NAME} DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/${STORAGE
 
 set(PLUGIN_COBALT_AUTOSTART false CACHE STRING "Automatically start Cobalt plugin")
 set(PLUGIN_COBALT_OUTOFPROCESS false CACHE STRING "Controls if the plugin should run in its own process")
+set(PLUGIN_COBALT_RESOLUTION "720p" CACHE STRING "Browser resolution")
+
+# resolution handling
+if(PLUGIN_COBALT_RESOLUTION EQUAL "720p")
+    set(PLUGIN_COBALT_HEIGHT "720")
+    set(PLUGIN_COBALT_WIDTH "1280")
+elseif(PLUGIN_COBALT_RESOLUTION EQUAL "1080p")
+    set(PLUGIN_COBALT_HEIGHT "1080")
+    set(PLUGIN_COBALT_WIDTH "1920")
+elseif(PLUGIN_COBALT_RESOLUTION EQUAL "2160p")
+    set(PLUGIN_COBALT_HEIGHT "2160")
+    set(PLUGIN_COBALT_WIDTH "3840")
+endif()
 
 write_config(${PLUGIN_NAME})
diff --git a/Cobalt/Cobalt.config b/Cobalt/Cobalt.config
index 17f042f..2b47424 100644
--- a/Cobalt/Cobalt.config
+++ b/Cobalt/Cobalt.config
@@ -8,8 +8,8 @@ ans(rootobject)
 
 map()
     kv(url "https://www.youtube.com/tv")
-    kv(height 720)
-    kv(width 1280)
+    kv(height ${PLUGIN_COBALT_HEIGHT})
+    kv(width ${PLUGIN_COBALT_WIDTH})
     if(PLUGIN_COBALT_CLIENTIDENTIFIER)
         kv(clientidentifier ${PLUGIN_COBALT_CLIENTIDENTIFIER})
     endif()
