diff --git a/DeviceInfo/DeviceInfo.cpp b/DeviceInfo/DeviceInfo.cpp
index f60f482..1398375 100644
--- a/DeviceInfo/DeviceInfo.cpp
+++ b/DeviceInfo/DeviceInfo.cpp
@@ -139,7 +139,7 @@ namespace Plugin {
             JsonData::DeviceInfo::AddressesData& element(addressInfo.Add(newElement));
 
             // get an interface with a public IP address, then we will have a proper MAC address..
-            Core::IPV4AddressIterator selectedNode(interfaces.Index());
+            Core::IPV4AddressIterator selectedNode(interfaces.IPV4Addresses());
 
             while (selectedNode.Next() == true) {
                 Core::JSON::String nodeName;
