diff --git a/cmake/modules/FindNXCLIENT.cmake b/cmake/modules/FindNXCLIENT.cmake
index 3315673..9e420c5 100644
--- a/cmake/modules/FindNXCLIENT.cmake
+++ b/cmake/modules/FindNXCLIENT.cmake
@@ -26,15 +26,23 @@
 # OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
 # ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
-find_path(LIBNXCLIENT_INCLUDE nexus_config.h
+find_path(NXCLIENT_INCLUDE nexus_config.h
         PATH_SUFFIXES refsw)
 
+find_path(NXCLIENT_NXSERVER_INCLUDE nxserverlib.h
+        PATH_SUFFIXES nxclient)
+
 find_library(LIBNXCLIENT_LIBRARY nxclient)
 
 if(EXISTS "${LIBNXCLIENT_LIBRARY}")
     include(FindPackageHandleStandardArgs)
 
     set(NXCLIENT_FOUND TRUE)
+    set(LIBNXCLIENT_INCLUDE ${NXCLIENT_INCLUDE})
+
+    if(EXISTS "${NXCLIENT_NXSERVER_INCLUDE}")
+       set(LIBNXCLIENT_INCLUDE ${NXCLIENT_NXSERVER_INCLUDE})
+    endif()
 
     find_package_handle_standard_args(LIBNXCLIENT DEFAULT_MSG NXCLIENT_FOUND LIBNXCLIENT_INCLUDE LIBNXCLIENT_LIBRARY)
     mark_as_advanced(LIBNXCLIENT_LIBRARY)
