From 38d616ed8a75f0b150049b401e71f2100832d7cb Mon Sep 17 00:00:00 2001
From: Pawel Stanek <p.stanek@metrological.com>
Date: Thu, 10 Oct 2019 13:45:04 +0200
Subject: [PATCH] [OCDM] Remove redundant API.

---
 Source/interfaces/IDRM.h | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/Source/interfaces/IDRM.h b/Source/interfaces/IDRM.h
index c94508f9..339a443e 100644
--- a/Source/interfaces/IDRM.h
+++ b/Source/interfaces/IDRM.h
@@ -354,11 +354,6 @@ class IMediaKeysExt {
 
     virtual uint64_t GetDrmSystemTime() const = 0;
 
-    // Destroy a MediaKeySession instance.
-    virtual CDMi_RESULT DestroyMediaKeySessionExt(
-        IMediaKeySession* f_piMediaKeySession)
-        = 0;
-
     virtual std::string GetVersionExt() const = 0;
 
     virtual uint32_t GetLdlSessionLimit() const = 0;
