From 652f6e0fbb2a27a13eb7d041479cb0ee45b0efcf Mon Sep 17 00:00:00 2001
From: Bram Oosterhuis <mail@bybram.com>
Date: Wed, 8 Jul 2020 14:19:19 +0200
Subject: [PATCH] bluetooth: fix error: narrowing conversion

---
 Source/bluetooth/HCISocket.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Source/bluetooth/HCISocket.h b/Source/bluetooth/HCISocket.h
index d69dec96..3149dd23 100644
--- a/Source/bluetooth/HCISocket.h
+++ b/Source/bluetooth/HCISocket.h
@@ -598,7 +598,7 @@ namespace Bluetooth {
 
     public:
 
-        template<const uint16_t OPCODE, typename OUTBOUND, typename INBOUND, const uint8_t RESPONSECODE = ~0>
+        template<const uint16_t OPCODE, typename OUTBOUND, typename INBOUND, const uint8_t RESPONSECODE = static_cast<uint8_t>(~0)>
         class CommandType : public Core::IOutbound, public Core::IInbound {
         private:
             CommandType<OPCODE, OUTBOUND, INBOUND, RESPONSECODE>& operator=(const CommandType<OPCODE, OUTBOUND, INBOUND, RESPONSECODE>&) = delete;
