From d815cea59e62b7995c94e2322a21e1aba89277bc Mon Sep 17 00:00:00 2001
From: Bram Oosterhuis <mail@bybram.com>
Date: Wed, 8 Jul 2020 14:17:32 +0200
Subject: [PATCH] websocket: fixes deduced conflicting types error on gcc 9.3

---
 Source/websocket/WebTransfer.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Source/websocket/WebTransfer.h b/Source/websocket/WebTransfer.h
index 99093df0..c59db3a4 100644
--- a/Source/websocket/WebTransfer.h
+++ b/Source/websocket/WebTransfer.h
@@ -336,7 +336,7 @@ namespace Web {
 
             // Read all Data to calculate the HASH/HMAC
             while (size > 0) {
-                uint16_t chunk = std::min(sizeof(buffer), size);
+                uint16_t chunk = std::min(static_cast<uint32_t>(sizeof(buffer)), size);
                 _fileBody.Serialize(buffer, chunk);
                 hash.Input(buffer, chunk);
                 size -= chunk;
@@ -502,7 +502,7 @@ namespace Web {
             uint8_t   buffer[64];
             typename ACTUALFILEBODY::HashType& hash = _fileBody.Hash();
             uint32_t  pos  = _fileBody.Position();
-            uint32_t  size = _fileBody.Size() - pos;
+            std::size_t  size = _fileBody.Size() - pos;
 
             hash.Reset();
 