From a9af19cde6a064d7572f4343b527656527086096 Mon Sep 17 00:00:00 2001
From: wouterlucas <wouter@wouterlucas.com>
Date: Wed, 23 May 2018 21:17:55 -0700
Subject: [PATCH] make graphic libs configurable

---
 src/dispatch_common.c | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/src/dispatch_common.c b/src/dispatch_common.c
index a38c0fc..b5ce446 100644
--- a/src/dispatch_common.c
+++ b/src/dispatch_common.c
@@ -173,12 +173,15 @@
 
 #include "dispatch_common.h"
 
+#define xstr(a) str(a)
+#define str(a) #a
+
 #ifdef __APPLE__
 #define GLX_LIB "/opt/X11/lib/libGL.1.dylib"
 #elif defined(ANDROID)
 #define GLX_LIB "libGLESv2.so"
 #else
-#define GLX_LIB "libGL.so.1"
+#define GLX_LIB xstr(GLX_LIB_NAME)
 #endif
 
 #ifdef ANDROID
@@ -190,9 +193,9 @@
 #define GLES1_LIB "libGLES_CM.dll"
 #define GLES2_LIB "libGLESv2.dll"
 #else
-#define EGL_LIB "libEGL.so.1"
-#define GLES1_LIB "libGLESv1_CM.so.1"
-#define GLES2_LIB "libGLESv2.so.2"
+#define EGL_LIB xstr(EGL_LIB_NAME)
+#define GLES1_LIB xstr(GLES1_LIB_NAME)
+#define GLES2_LIB xstr(GLES2_LIB_NAME)
 #endif
 
 #ifdef __GNUC__
-- 
2.17.0

