From 1861041379a7cccc758bd54d845e1622928eabd7 Mon Sep 17 00:00:00 2001
From: Bram Oosterhuis <mail@bybram.com>
Date: Sat, 27 Mar 2021 13:38:15 +0100
Subject: [PATCH] cmake: become more easy in findgbm

---
 cmake/FindGBM.cmake | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/cmake/FindGBM.cmake b/cmake/FindGBM.cmake
index 9270f1d..3888b2c 100644
--- a/cmake/FindGBM.cmake
+++ b/cmake/FindGBM.cmake
@@ -26,7 +26,7 @@ if(${PKG_CONFIG_FOUND})
 
     # Just check if the gbm.pc exist, and create the PkgConfig::gbm target
     # No version requirement (yet)
-    pkg_check_modules(GBM REQUIRED IMPORTED_TARGET gbm)
+    pkg_check_modules(PC_GBM REQUIRED IMPORTED_TARGET gbm)
 
     include(FindPackageHandleStandardArgs)
 
@@ -34,17 +34,17 @@ if(${PKG_CONFIG_FOUND})
     find_package_handle_standard_args(
         gbm
         REQUIRED_VARS
-            GBM_INCLUDE_DIRS
-            GBM_CFLAGS
-            GBM_LDFLAGS
-            GBM_LIBRARIES
-            GBM_LIBRARY_DIRS
+            PC_GBM_LIBRARIES
         VERSION_VAR
             GBM_VERSION
     )
 
     find_library(GBM_ACTUAL_LIBRARY NAMES gbm 
-        HINTS ${GBM_LIBRARY_DIRS} )
+        HINTS ${PC_GBM_LIBRARY_DIRS} )
+    
+    find_path(GBM_INCLUDE_DIR NAMES gbm.h
+        HINTS ${PC_GBM_INCLUDEDIR} ${PC_GBM_INCLUDE_DIRS}
+    )
 
     if(GBM_FOUND AND NOT TARGET libgbm::libgbm)
         add_library(libgbm::libgbm UNKNOWN IMPORTED)
@@ -52,7 +52,7 @@ if(${PKG_CONFIG_FOUND})
             IMPORTED_LOCATION "${GBM_ACTUAL_LIBRARY}"
             INTERFACE_LINK_LIBRARIES "${GBM_LIBRARIES}"
             INTERFACE_COMPILE_OPTIONS "${GBM_CFLAGS}"
-            INTERFACE_INCLUDE_DIRECTORIES "${GBM_INCLUDE_DIRS}"
+            INTERFACE_INCLUDE_DIRECTORIES "${GBM_INCLUDE_DIR}"
             )
     else()
         message(FATAL_ERROR "Some required variable(s) is (are) not found / set! Does gbm.pc exist?")
-- 
2.25.1

