From 7ff652bf192ad7293df83bd86aad85e8ea0fd7b4 Mon Sep 17 00:00:00 2001
From: Andre McCurdy <armccurdy@gmail.com>
Date: Wed, 18 Nov 2015 18:56:54 -0800
Subject: [PATCH] WTF: disable use of dladdr with musl libc

Signed-off-by: Andre McCurdy <armccurdy@gmail.com>
---
 Source/WTF/wtf/Assertions.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Source/WTF/wtf/Assertions.cpp b/Source/WTF/wtf/Assertions.cpp
index 44d8641..c7f8435 100644
--- a/Source/WTF/wtf/Assertions.cpp
+++ b/Source/WTF/wtf/Assertions.cpp
@@ -265,7 +265,7 @@ void WTFReportBacktrace()
 #    if defined(__GLIBC__) && !defined(__UCLIBC__)
 #      define USE_BACKTRACE_SYMBOLS 1
 #    endif
-#  elif !defined(__UCLIBC__)
+#  elif defined(__GLIBC__) && !defined(__UCLIBC__)
 #    define USE_DLADDR 1
 #  endif
 #endif
-- 
1.9.1

