Index: git/Source/JavaScriptCore/heap/MachineStackMarker.cpp
===================================================================
--- git.orig/Source/JavaScriptCore/heap/MachineStackMarker.cpp
+++ git/Source/JavaScriptCore/heap/MachineStackMarker.cpp
@@ -539,7 +539,7 @@ void* MachineThreads::Thread::Registers:
 #error Unknown Architecture
 #endif
 
-#elif defined(__GLIBC__) && ENABLE(JIT)
+#elif OS(LINUX) && ENABLE(JIT)
 
 #if CPU(X86)
     return reinterpret_cast<void*>((uintptr_t) regs.machineContext.gregs[REG_ESP]);
@@ -638,7 +638,7 @@ void* MachineThreads::Thread::Registers:
 #error Unknown Architecture
 #endif
 
-#elif defined(__GLIBC__)
+#elif OS(LINUX) // glibc and musl
 
 // The following sequence depends on glibc's sys/ucontext.h.
 #if CPU(X86)
@@ -720,7 +720,7 @@ void* MachineThreads::Thread::Registers:
 #error Unknown Architecture
 #endif
 
-#elif defined(__GLIBC__)
+#elif OS(LINUX) // glibc and musl
 
 // The following sequence depends on glibc's sys/ucontext.h.
 #if CPU(X86)
@@ -811,7 +811,7 @@ void* MachineThreads::Thread::Registers:
 #error Unknown Architecture
 #endif
 
-#elif defined(__GLIBC__)
+#elif OS(LINUX) // glibc and musl
 
 // The following sequence depends on glibc's sys/ucontext.h.
 #if CPU(X86)
