From 28202ceb25dcfc1fc9c71b2fc93c61776896bc45 Mon Sep 17 00:00:00 2001
From: wouterlucas <wouter@wouterlucas.com>
Date: Mon, 19 Mar 2018 14:17:50 -0700
Subject: [PATCH] [cmake] Remove redundant include

---
 CMakeLists.txt | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4007593..dd1d225 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -79,10 +79,6 @@ endif()
 
 set(PLUGIN_ROOT "${CMAKE_CURRENT_SOURCE_DIR}")
 
-if(CMAKE_CROSSCOMPILING)
-    include(${CMAKE_FIND_ROOT_PATH}/usr/include/WPEFramework/cmake/config.cmake)
-endif(CMAKE_CROSSCOMPILING)
-
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
 list(APPEND CMAKE_MODULE_PATH "${WPEFRAMEWORK_PLUGIN_DIAL_ROOT}/cmake")
 
-- 
2.11.0

