diff -Nuar a/Makefile.in b/Makefile.in
--- a/Makefile.in	2020-01-22 06:57:59.702910660 +0100
+++ b/Makefile.in	2020-01-22 06:57:37.923999218 +0100
@@ -1802,7 +1802,7 @@
 	$(systemduserunit_DATA)
 am__include_HEADERS_DIST = lib/bluetooth.h lib/hci.h lib/hci_lib.h \
 	lib/sco.h lib/l2cap.h lib/sdp.h lib/sdp_lib.h lib/rfcomm.h \
-	lib/bnep.h lib/cmtp.h lib/hidp.h
+	lib/bnep.h lib/cmtp.h lib/hidp.h lib/mgmt.h
 HEADERS = $(include_HEADERS)
 am__tagged_files = $(HEADERS) $(SOURCES) $(TAGS_FILES) \
 	$(LISP)config.h.in
@@ -2272,7 +2272,7 @@
 lib_sources = lib/bluetooth.c lib/hci.c lib/sdp.c
 lib_headers = lib/bluetooth.h lib/hci.h lib/hci_lib.h \
 		lib/sco.h lib/l2cap.h lib/sdp.h lib/sdp_lib.h \
-		lib/rfcomm.h lib/bnep.h lib/cmtp.h lib/hidp.h
+		lib/rfcomm.h lib/bnep.h lib/cmtp.h lib/hidp.h lib/mgmt.h
 
 extra_headers = lib/mgmt.h lib/uuid.h lib/a2mp.h lib/amp.h
 extra_sources = lib/uuid.c
